phenol-annotations/src/main/java/org/monarchinitiative/phenol/annotations/assoc/GeneInfoLoader.java

```diff
@@ -0,0 +1,52 @@
+package org.monarchinitiative.phenol.annotations.assoc;
+
+import org.monarchinitiative.phenol.annotations.formats.GeneIdentifier;
+import org.monarchinitiative.phenol.annotations.formats.GeneIdentifiers;
+import org.monarchinitiative.phenol.ontology.data.TermId;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.*;
+
+public class GeneInfoLoader {
+  private static final String ENTREZ_GENE_PREFIX = "NCBIGene";
+
+
+  private GeneInfoLoader() {
+    // should not be instantiated
+  }
+
+  /**
+   * @param geneInfoFile Homo_sapiens.gene_info.gz File
+   * @return an immutable Map with keys being NCBI Gene IDs and values being the corresponding symbols
+   */
+  public static GeneIdentifiers loadGeneIdentifiers(Path geneInfoFile) throws IOException {
+    List<GeneIdentifier> builder = new ArrayList<>();
+    try (BufferedReader reader = FileUtils.openForReading(geneInfoFile)) {
+      String line;
+      // We have seen that occasionally the Homo_sapiens_gene_info.gz
+      // contains duplicate lines, which is an error but we do not want the code
+      // to crash, so we check for previously found term ids with the seen set.
+      // The TermId <-> symbol mapping is one to one.
+      Set<TermId> seen = new HashSet<>();
+      while ((line = reader.readLine()) != null) {
+        String[] a = line.split("\t");
```

Comment:
Not sure if this could be neater done as a map, filter, distinct on a stream e.g.

```java
Files.lines(geneInfoFile)
.map(toGeneIdentifier)
.filter(...)
.distinct()
.collect(Collectors.toUnmodifiableList());
```

Reply:
Done, thanks!